app/scripts/controllers/detect-tokens.test.js

```diff
@@ -559,4 +559,33 @@ describe('DetectTokensController', function () {
     clock.tick(180000);
     sandbox.assert.notCalled(stub);
   });
+
+  it('should poll on the correct interval by networkClientId', async function () {
+    const clock = sandbox.useFakeTimers();
+    const controller = new DetectTokensController({
+      messenger: getRestrictedMessenger(),
+      preferences,
+      network,
+      tokensController,
+      assetsContractController,
+      getNetworkClientById: () => ({
+        configuration: {
+          chainId: '0x1',
+        },
+        provider: {},
+        blockTracker: {},
+        destroy: () => {
+          // noop
+        },
+      }),
+    });
+    const stub = sandbox.stub(controller, 'detectNewTokens').resolves('foo');
+    controller.startPollingByNetworkClientId('mainnet');
+    clock.tick(180000);
+    await Promise.resolve();
```

Comment:
You probably want to use `tickAsync` here instead:

```suggestion
    clock.tickAsync(180000);
```

`tick` will wait synchronously, so it won't clear the promise queue.